### ISSUE TITLE
Problem with installation

### ISSUE THREAD

**Issue by dlmocdm** [premgopalan/hgaprec#7]
I tried to install this on both ubuntu and mac os computers, but I always get this error when executing "make":

Making all in src
make[1]: Entering directory `/home/dlmocdm/hgaprec-master2/src'
source='ratings.cc' object='ratings.o' libtool=no \
    DEPDIR=.deps depmode=none /bin/bash ../depcomp \
    g++ -DPACKAGE_NAME=\"hgaprec\" -DPACKAGE_TARNAME=\"hgaprec\" -DPACKAGE_VERSION=\"0.1\" -DPACKAGE_STRING=\"hgaprec\ 0.1\" -DPACKAGE_BUGREPORT=\"pgopalan@cs.princeton.edu\" -DPACKAGE=\"hgaprec\" -DVERSION=\"0.1\" -DHAVE_LIBGSLCBLAS=1 -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGSL=1 -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDINT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_SYS_FILE_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_UNISTD_H=1 -DHAVE__BOOL=1 -DHAVE_STDBOOL_H=1 -DHAVE_STDLIB_H=1 -DHAVE_MALLOC=1 -DHAVE_GETTIMEOFDAY=1 -DHAVE_LOCALTIME_R=1 -DHAVE_MEMSET=1 -DHAVE_MKDIR=1 -DHAVE_STRERROR=1 -DHAVE_STRTOL=1 -I.   -I/opt/local/include  -O3 -c -o ratings.o ratings.cc
/bin/bash: ../depcomp: No such file or directory
make[1]: *** [ratings.o] Error 127
make[1]: Leaving directory`/home/dlmocdm/hgaprec-master2/src'
make: **\* [all-recursive] Error 1

Any idea how can I solve this?

Thanks!

**Comment by lcharlin** [comment]
Have you tried:

automake --add-missing

(I believe Prem suggested that last time this came up)

**Comment by dlmocdm** [comment]
Thanks. In the end, I was missing some gsl libraries. However, now I get a new error when I execute "make":

Making all in src
make[1]: Entering directory `/home/dlmocdm/hgaprec-master/src'
make[1]: Nothing to be done for`all'.
make[1]: Leaving directory `/home/dlmocdm/hgaprec-master/src'
make[1]: Entering directory`/home/dlmocdm/hgaprec-master'
make[1]: Nothing to be done for `all-am'.
make[1]: Leaving directory`/home/dlmocdm/hgaprec-master'

Automake doesn't solve this... :/

**Comment by lcharlin** [comment]
In the log you sent it appears that everything was compiled as expected.

You should now be able to run the program.

**Comment by dlmocdm** [comment]
No, I cannot run it. When I try to execute the Pearl script according to README, I get the following:
_sh: 1: /home/dlmocdm/hgaprec-master/src: Permission denied_

**Comment by dlmocdm** [comment]
Thanks, it worked! But... one more question... 
When I execute the Pearl script: _./scripts/run.pl -dataset movielens -binary_ 
where is the output stored?

Thanks!

**Comment by submagr** [comment]
I tried installing hgaprec. After ./configure and automake --add-missing, I tried make. But I'm getting the following error: 

`Making all in src
make[1]: Entering directory '~/hgaprec/src'
g++  -O3   -o hgaprec ratings.o main.o log.o hgaprec.o -larpack -llapack -lblas -lgsl -lpthread -lgslcblas 
/usr/bin/ld: cannot find -larpack
collect2: error: ld returned 1 exit status
Makefile:233: recipe for target 'hgaprec' failed
make[1]: *** [hgaprec] Error 1
make[1]: Leaving directory '~/hgaprec/src'
Makefile:237: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
`

I tried to google about larpack, but I didn't find anywhere about it. 
Any ideas on how to solve this ??

**Comment by slinnarsson** [comment]
See issue #9 for how to install on Mac.
